lhotse/recipes/aishell.py

```diff
@@ -74,46 +77,47 @@ def prepare_aishell(
             transcript_dict[idx_transcript[0]] = ' '.join(idx_transcript[1:])
     manifests = defaultdict(dict)
     dataset_parts = ['train', 'dev', 'test']
-    for part in dataset_parts:
-        # Generate a mapping: utt_id -> (audio_path, audio_info, speaker, text)
-        recordings = []
-        supervisions = []
-        wav_path = corpus_dir / 'data_aishell' / 'wav' / f'{part}'
-        for audio_path in wav_path.rglob('**/*.wav'):
-            idx = audio_path.stem
-            speaker = audio_path.parts[-2]
-            if idx not in transcript_dict:
-                logging.warning(f'No transcript: {idx}')
-                continue
-            text = transcript_dict[idx]
-            if not audio_path.is_file():
-                logging.warning(f'No such file: {audio_path}')
-                continue
-            recording = Recording.from_file(audio_path)
-            recordings.append(recording)
-            segment = SupervisionSegment(
-                id=idx,
-                recording_id=idx,
-                start=0.0,
-                duration=recording.duration,
-                channel=0,
-                language='Chinese',
-                speaker=speaker,
-                text=text.strip()
-            )
-            supervisions.append(segment)
+    with ThreadPoolExecutor(num_jobs) as ex:
```

Comment:
The `ex` object is never used here, so nothing actually gets processed in parallel. You’d have to use either `ex.submit` or `ex.map` to leverage the thread pool (I think there’s an example in the librispeech recipe).

Reply:
ok. thank you.